Source/AwsCommonRuntimeKit/crt/CommonRuntimeError.swift

```diff
@@ -23,6 +23,20 @@ public struct CRTError: Equatable {
         self.name = String(cString: aws_error_name(self.code))
     }
 
+    public init<T: BinaryInteger>(code: T, context: String?) {
+        if code > INT32_MAX || code <= 0 {
+            self.code = Int32(AWS_ERROR_UNKNOWN.rawValue)
```

Comment:
Is it intentional that `code == 0` is treated as unknown error? It doesn't feel right, because `AWS_ERROR_SUCCESS` is valid code and can be passed to this function, even if by accident (e.g. by `aws_last_error ()` call).

Reply:
Yes, that was intentional. This can lead a bad call to last error becoming AWS_ERROR_SUCCESS, which will probably make the program fail at some later stage, making it harder to debug. I think it is easier to debug where the unknown error is coming from and fix that.